### ISSUE TITLE
fix: coinsurance percentage from InsuranceConfig never applied in claim processing

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#1235]
## Problem Statement

`InsuranceConfig` defines a `coinsurance` field (line 72 of `config/insurance.py`) with a valid range of (0, 1], defaulting to 1.0. However, this coinsurance percentage is **never applied** anywhere in the claim processing pipeline. Neither `process_insurance_claim()` in `manufacturer_claims.py` nor `InsuranceProgram.process_claim()` in `insurance_program.py` references the coinsurance factor.

In real insurance programs, coinsurance clauses require the insured to bear a percentage of each covered loss (e.g., 80/20 coinsurance means the insurer pays 80% and the insured retains 20% of the covered amount above the deductible). This is distinct from the deductible and is a standard feature of commercial property and health insurance programs.

## v1.0 Impact

**Medium** — The config field is misleading because users can set `coinsurance=0.80` expecting 80% coverage, but 100% of the covered loss (up to the limit) is always recovered. This overstates insurance recovery and understates the company's retained loss in all simulations that set coinsurance < 1.0.

## Affected Code

- `ergodic_insurance/config/insurance.py:72` — `coinsurance` field defined but unused
- `ergodic_insurance/manufacturer_claims.py:238-438` — `process_insurance_claim()` does not apply coinsurance
- `ergodic_insurance/insurance_program.py` — `process_claim()` does not reference coinsurance

## Current Behavior

```python
# User sets coinsurance to 80%
config = InsuranceConfig(coinsurance=0.80, ...)
# Claim of $100K with $10K deductible, $1M limit:
# Expected: company pays $10K + 20% * $90K = $28K, insurer pays $72K
# Actual:   company pays $10K, insurer pays $90K (coinsurance ignored)
```

## Expected Behavior

Insurance recovery should be multiplied by the coinsurance factor:

```python
insured_loss = min(claim - deductible, limit)
insurance_payment = insured_loss * coinsurance  # Apply coinsurance
company_payment = claim - insurance_payment
```

## Recommended Approach

Apply coinsurance factor in `process_insurance_claim()` between the deductible calculation and the limit cap:
1. Calculate insured amount above deductible
2. Apply coinsurance percentage
3. Then apply limit cap to the coinsured amount
4. Ensure LAE calculations use the correct base amounts

## Acceptance Criteria

- [ ] `process_insurance_claim()` applies coinsurance factor from config
- [ ] `InsuranceProgram.process_claim()` respects coinsurance if configured
- [ ] Unit tests verify coinsurance reduces insurance recovery proportionally
- [ ] Coinsurance of 1.0 (default) produces identical results to current behavior
